locator/guice/src/main/java/org/eclipse/kapua/locator/guice/GuiceLocatorImpl.java

```diff
@@ -129,230 +133,64 @@ private void init(String locatorConfigName) throws Exception {
             throw new Exception("Locator configuration cannot be found: " + locatorConfigName);
         }
 
-        LocatorConfig locatorConfig = LocatorConfig.fromURL(locatorConfigURL);
-
+        final LocatorConfig locatorConfig = LocatorConfig.fromURL(locatorConfigURL);
+        ConfigurationPrinter configurationPrinter =
+                ConfigurationPrinter
+                        .create()
+                        .withLogger(LOG)
+                        .withLogLevel(ConfigurationPrinter.LogLevel.INFO)
+                        .withTitle("Kapua Locator Configuration")
+                        .addParameter("Resource Name", locatorConfigURL.getPath());
         //
         // Scan packages listed in to find KapuaModules
-        Collection<String> packageNames = locatorConfig.getIncludedPackageNames();
-        Reflections reflections = new Reflections(packageNames);
-        Set<Class<? extends AbstractKapuaModule>> kapuaModuleClasses = reflections.getSubTypesOf(AbstractKapuaModule.class);
+        final Reflections reflections = new Reflections(locatorConfig.getIncludedPackageNames());
+        final Set<Class<? extends AbstractKapuaModule>> kapuaModuleClasses = reflections.getSubTypesOf(AbstractKapuaModule.class);
 
         //
         // Instantiate Kapua modules
-        List<AbstractKapuaModule> kapuaModules = new ArrayList<>();
-        List<Class<? extends AbstractKapuaModule>> excludedKapuaModules = new ArrayList<>();
-        for (Class<? extends AbstractKapuaModule> moduleClazz : kapuaModuleClasses) {
-            if (isExcluded(moduleClazz.getName(), locatorConfig.getExcludedPackageNames())) {
-                excludedKapuaModules.add(moduleClazz);
-                continue;
-            }
+        final Map<Boolean, List<Class<? extends AbstractKapuaModule>>> includedAndExcludedModules = kapuaModuleClasses
+                .stream()
+                .collect(Collectors.partitioningBy(moduleClazz -> !locatorConfig.getExcludedPackageNames().stream().anyMatch(pn -> moduleClazz.getName().startsWith(pn))));
 
-            kapuaModules.add(moduleClazz.newInstance());
+        final List<AbstractKapuaModule> instantiatedKapuaModules = new ArrayList<>();
+        for (Class<? extends AbstractKapuaModule> moduleClazz : includedAndExcludedModules.get(true)) {
+            instantiatedKapuaModules.add(moduleClazz.newInstance());
         }
 
         // KapuaModule will be removed as soon as bindings will be moved to local modules
-        kapuaModules.add(new KapuaModule(locatorConfigName));
-
-        //
-        // Print loaded stuff
-        printLoadedKapuaModuleConfiguration(locatorConfigURL, locatorConfig, kapuaModules, excludedKapuaModules);
-
-        //
-        // Scan XmlSerializable
-        Set<Class<?>> xmlSerializableClasses = reflections.getTypesAnnotatedWith(XmlRootElement.class);
```

Comment:
Separation of concerns!!!! Class path scan is now isolated in its own class, instead of being thrown in the middle of this